velox/functions/sparksql/String.h

```diff
@@ -1181,4 +1181,140 @@ struct FindInSetFunction {
   }
 };
 
+template <typename T>
+struct LevenshteinDistanceFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  void call(
+      out_type<int32_t>& result,
+      const arg_type<Varchar>& left,
+      const arg_type<Varchar>& right,
+      int32_t threshold) {
+    auto leftCodePoints = stringImpl::stringToCodePoints(left);
+    auto rightCodePoints = stringImpl::stringToCodePoints(right);
+    doCall<int32_t>(
+        result,
+        leftCodePoints.data(),
+        rightCodePoints.data(),
+        leftCodePoints.size(),
+        rightCodePoints.size(),
+        threshold);
+  }
+
+  void callAscii(
+      out_type<int32_t>& result,
+      const arg_type<Varchar>& left,
+      const arg_type<Varchar>& right,
+      int32_t threshold) {
+    auto leftCodePoints = reinterpret_cast<const uint8_t*>(left.data());
+    auto rightCodePoints = reinterpret_cast<const uint8_t*>(right.data());
+    doCall<uint8_t>(
```

Comment:
I notice we are using `doCall<uint8_t>` and `doCall<int32_t>`, and what's the reason to have different handling? Is it possible to keep one way and remove the template parameter? Thanks.

Reply:
They are for handling unicode characters for exmaple we need to use multi chars to store chinese characters , but 1 char to store ascii characters.

Reply:
I reference the impl here: https://github.com/facebookincubator/velox/blob/main/velox/functions/prestosql/StringFunctions.h#L409C8-L409C36

Reply:
Got it. Thanks.